cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/input/BasicPartitionedFileSetInputContext.java

```diff
@@ -80,7 +127,7 @@ private PartitionKey getPartitionKey(URI inputPathURI) {
     }
     throw new IllegalArgumentException(
       String.format("Failed to derive PartitionKey from input path '%s' and path to key mapping '%s'.",
-                    inputPath, pathToPartitionMapping));
+                    inputPathURI, pathToPartitionMapping));
```

Comment:
This should only happen in some rare cases, where there's a bug or unexpected path encoding. If we don't log it, then its hard to debug in case of such a failure.
Perhaps we at least log the first 200 entries, so that in most cases, it logs all the useful information, and only trims if the map is too large?